Readme.md

```diff
@@ -1,87 +1,153 @@
 
-Based on SQlite version 3.40.1.
+Sqlite fork adapted to work with Fluence protocol. The current version is based on SQlite version 3.40.1. The artifact is sqlite WASM module:
+
+- that is compiled following [Marine runtime ABI conventions](https://fluence.dev/docs/marine-book/marine-rust-sdk/module-abi)
+- to be used with [Marine runtime](https://github.com/fluencelabs/marine)
+
 
 # How to build
 
-This app could be built either with docker
+
+
+Wasm module can be built with either docker-compose
+
+
 
 ```bash
+
 docker-compose up
+
 ```
 
-or by Makefile with [wasi-sdk](https://github.com/WebAssembly/wasi-sdk) installed
+
+
+or using GNU make. There are prerequisites to be installed following this path, namely [wasi-sdk](https://github.com/WebAssembly/wasi-sdk) and marine crate. Take a look at the Dockerfile for the details.
+
 ```bash
+
 make
+
 ```
 
+
+
 # How to use
 
-At now, this fork exports five API functions:
-```cpp
-
-/**
- * Executes given SQL request and returns result in as a pointer to the following structure: | result size (4 bytes, le)| result (size bytes) |.
- *
- * @param sql a pointer to the supplied sql request
- * @param length a size of the supplied sql request
- * @return a pointer to the struct contains result_size and result
- */
-char *invoke(const char *sql, size_t length);
-
-/**
- * Allocates a memory region of a given size. Could be used by Wasm execution environments for byte array passing.
- *
- * @param size a size of allocated memory region
- * @return a pointer to the allocated memory region
- */
-void *allocate(size_t size);
-
-/**
- * Frees a memory region. Could be used by Wasm execution environments for freeing previous memory allocated by `allocate` function.
- *
- * @param ptr a pointer to the previously allocated memory region
- * @param size a size of the previously allocated memory region
- */
-void deallocate(void *ptr, size_t size);
-
-/**
- * Stores one byte by a given address in the module memory.
- *
- * @param ptr a address where byte should be stored
- * @param value a byte to be stored
- */
-void store(void *ptr, char value);
-
-/**
- * Returns one byte by a given address in the module memory.
- *
- * @param ptr a address at which the needed byte is located
- * @return the byte at the given address
- */
-char load(void *ptr);
+The Sqlite Wasm module exports a set of Sqlite C API functions.  There are multiple ways to try these Wasm module functions.
+The fastest way tp try is to run it using a combination of marine and mrepl Rust crates.
+```bash
+/git/sqlite-wasm-connector (master)$cargo install marine mrepl
+...
+/git/sqlite-wasm-connector (master)$marine repl Config.toml
+```
+You can find MREPL output for a simple "CREATE-INSERT-SELECT" scenario below. Mind the second argument to sqlite3_open_v2() that is an OR-ed Sqlite flag values. 6 in this context is SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE.
```

Comment:
Let's wrap with backticks ` SQLITE_OPEN_READWRITE and SQLITE_OPEN_CREATE

Reply:
Done